File: Python/2006.py
# 2006. Count Number of Pairs With Absolute Difference K
# https://leetcode.com/problems/count-number-of-pairs-with-absolute-difference-k/

class Solution:
  def countKDifference(self, nums: List[int], k: int) -> int:
    length = len(nums)
    out = 0

    for i in range(0, length - 1):
      for j in range(i + 1, length):
        if (abs(nums[i] - nums[j]) == k):
          out += 1

    return out

File: Rust/2006.rs
/**
 * 2006. Count Number of Pairs With Absolute Difference K
 * https://leetcode.com/problems/count-number-of-pairs-with-absolute-difference-k/
 */

impl Solution {
    pub fn count_k_difference(nums: Vec<i32>, k: i32) -> i32 {
        let len: usize = nums.len();
        let mut out: i32 = 0;

        for i in 0..len-1 {
          for j in i+1..len {
            if ((nums[i]-nums[j]).abs()==k) {
              out += 1;
            }
          }
        }

        return out;
    }
}